cpp/src/arrow/type.h

```diff
@@ -147,6 +147,15 @@ struct Type {
   };
 };
 
+struct ARROW_EXPORT DataTypeLayout {
+  // The bit width for each buffer in this DataType's representation
+  // (-1 if the item size for a given buffer is unknown or variable,
+  //  0 if the buffer is always null).
+  // Child types are not included, they should be inspected separately.
+  std::vector<int64_t> bit_widths;
+  bool has_dictionary;
```

Comment:
Can you specify a default?

Reply:
I'd like to but C++ disables default constructors when you add a default initializer to a member field :-/